### PR TITLE
fix: updating telemetry docs and examples

### PR DESCRIPTION
**Pull request by brunos-bq**
### Summary

Latest changes to telemetry documentation prior to release

### Additional Reviewers

@sergiyv-improving 

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/using-the-jdbc-driver/Telemetry.md [hunk 1]
```diff
@@ -44,7 +44,30 @@ Before enabling the Telemetry feature, a few setup steps are required to ensure
 
 Telemetry for the AWS JDBC Driver is a monitoring strategy that overlooks all plugins enabled in [`wrapperPlugins`](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters) and is not a plugin in itself. Therefore no changes are required in the `wrapperPlugins` parameter to enable Telemetry.
 
-In order to enable Telemetry in the AWS JDBC Driver all you need to do is to set the `enableTelemetry` property to `true` either in the code or in the connection string.
+In order to enable Telemetry in the AWS JDBC Driver you need to first set the `enableTelemetry` property to `true` either in the code or in the connection string.
+
+After that, it will be needed to, in the application code that will be using the driver, to set up the recorders that will take care of exporting the telemetry data from the code to the ADOT Collector.
+
+Below you find an example on how to set up the OpenTelemetry SDK and its configurations:
```
Comment (crystall-bitquill) [inline docs/using-the-jdbc-driver/Telemetry.md:51]: Can a link to the example file be added here? And on the main examples table on the readme: 
 https://github.com/awslabs/aws-advanced-jdbc-wrapper#examples

**Review by sergiyvamz — APPROVED** [review]

File: docs/using-the-jdbc-driver/Telemetry.md [hunk 2]
```diff
@@ -44,7 +44,30 @@ Before enabling the Telemetry feature, a few setup steps are required to ensure
 
 Telemetry for the AWS JDBC Driver is a monitoring strategy that overlooks all plugins enabled in [`wrapperPlugins`](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters) and is not a plugin in itself. Therefore no changes are required in the `wrapperPlugins` parameter to enable Telemetry.
```
Comment (karenc-bq) [inline docs/using-the-jdbc-driver/Telemetry.md:45]: Can you please update the warning sections above to:
```
> [!WARNING]\
> The AWS Distro for OpenTelemetry Collector can be set up either locally or remotely. It is up to the user to decide where is best to set it up. If you decide to host it remotely, ensure that the application has the necessary permissions or allowlists to connect to the Collector.
> The collector is an external application that is not part of the wrapper itself. Without a collector, the wrapper will collect monitoring data from its execution but that data will not be sent anywhere for visualization.
```

I found this out the other day and I think it does a better job at drawing attention.
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

File: docs/using-the-jdbc-driver/Telemetry.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 Monitoring is the ability to gather data and insights on the execution of an application. Users will also be able to inspect the gathered data and determine potential actions to take depending on the data collected.
 
-In version X.Y.Z of the AWS JDBC Driver, we have introduced a Telemetry feature to the driver enabling users to collect and visualize data of the AWS JDBC Driver execution, at a global level and at plugin level, meaning that one can now inspect specific performance data for each plugin activated in the properties.
+In version 2.3.0 of the AWS JDBC Driver, we have introduced a Telemetry feature to the driver enabling users to collect and visualize data of the AWS JDBC Driver execution, at a global level and at plugin level, meaning that one can now inspect specific performance data for each plugin activated in the properties.
 
 This enables you to monitor the performance of the driver with your configurations and determine whether the performance meets your expectations on the driver.
```
Comment (karenc-bq) [inline docs/using-the-jdbc-driver/Telemetry.md:7]: Can we merge this sentence with the previous paragraph?

What do you think of something like 

```
Version 2.3.0 of the AWS JDBC Driver introduced the Telemetry feature. This feature allows you to collect and visualize data of the AWS JDBC Driver execution at a global level and at plugin level. You can now monitor the performance of the driver as a whole or within specific plugins with your configurations, and determine whether the driver's performance meets your expectations.

File: docs/using-the-jdbc-driver/Telemetry.md [hunk 4]
```diff
@@ -44,7 +44,30 @@ Before enabling the Telemetry feature, a few setup steps are required to ensure
```
Comment (karenc-bq) [inline docs/using-the-jdbc-driver/Telemetry.md:44]: In point 2 above can we link to AWS Distro for OpenTelemetry Collector's download page?

File: docs/using-the-jdbc-driver/Telemetry.md [hunk 5]
```diff
@@ -44,7 +44,30 @@ Before enabling the Telemetry feature, a few setup steps are required to ensure
 
 Telemetry for the AWS JDBC Driver is a monitoring strategy that overlooks all plugins enabled in [`wrapperPlugins`](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters) and is not a plugin in itself. Therefore no changes are required in the `wrapperPlugins` parameter to enable Telemetry.
 
-In order to enable Telemetry in the AWS JDBC Driver all you need to do is to set the `enableTelemetry` property to `true` either in the code or in the connection string.
+In order to enable Telemetry in the AWS JDBC Driver you need to first set the `enableTelemetry` property to `true` either in the code or in the connection string.
+
+After that, it will be needed to, in the application code that will be using the driver, to set up the recorders that will take care of exporting the telemetry data from the code to the ADOT Collector.
```
Comment (karenc-bq) [inline docs/using-the-jdbc-driver/Telemetry.md:49]: Can we convert this to bullet points?
1. Set the `enableTelemetry` property to `true`. You can either set it in the through `Properties` or through the connection string.
2. Set up the recorders that exports the telemetry data from the code to the ADOT Collector.

Comment (karenc-bq) [reply docs/using-the-jdbc-driver/Telemetry.md:49]: How do you set up the recorders? If it is the sample code below the relation is not very obvious on first read. Can we merge them into the same paragraph?

File: docs/using-the-jdbc-driver/Telemetry.md [hunk 6]
```diff
@@ -44,7 +44,30 @@ Before enabling the Telemetry feature, a few setup steps are required to ensure
...
+
+Below you find the piece of code required to set up the OpenTelemetry SDK and its configurations. The full code of the example on how to set up the AWS JDBC Driver using Telemetry can also be found in the examples folder of this repository. [Example code](examples/AWSDriverExample/src/main/java/software/amazon/TelemetryMetricsOTLPExample.java).
+
+```
```
Comment (aaronchung-bitquill) [inline docs/using-the-jdbc-driver/Telemetry.md:53]: Can you specify `java` ?

File: docs/using-the-jdbc-driver/Telemetry.md [hunk 7]
```diff
@@ -34,17 +32,48 @@ Before enabling the Telemetry feature, a few setup steps are required to ensure
...
 Telemetry for the AWS JDBC Driver is a monitoring strategy that overlooks all plugins enabled in [`wrapperPlugins`](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters) and is not a plugin in itself. Therefore no changes are required in the `wrapperPlugins` parameter to enable Telemetry.
 
-In order to enable Telemetry in the AWS JDBC Driver all you need to do is to set the `enableTelemetry` property to `true` either in the code or in the connection string.
+In order to enable Telemetry in the AWS JDBC Driver, you need to:
+
+1. Set the `enableTelemetry` property to `true`. You can either set it through Properties or directly in the connection string.
+
+2. Set up the recorders that will export the telemetry data from the code to the ADOT Collector.
+
+Setting up the recorders require to instantiate an `OpenTelemetrySDK` in the application code prior to executing the driver. Instantiating the `OpenTelemetrySDK` requires you to configure the endpoints where traces and metrics are being forwarded to,
```
Comment (karenc-bq) [inline docs/using-the-jdbc-driver/Telemetry.md:53]: This sentenced ended in a comma

File: docs/using-the-jdbc-driver/Telemetry.md [hunk 8]
```diff
@@ -34,17 +32,48 @@ Before enabling the Telemetry feature, a few setup steps are required to ensure
...
+    .build();
+
+OpenTelemetrySdk.builder()
+    .setTracerProvider(tracerProvider)
+    .setMeterProvider(meterProvider)
+    .setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
+    .buildAndRegisterGlobal();
+```
+
+We also provide a full example of a sample application using telemetry in the examples folder of this repository. [Example code](examples/AWSDriverExample/src/main/java/software/amazon/TelemetryMetricsOTLPExample.java).
```
Comment (karenc-bq) [inline docs/using-the-jdbc-driver/Telemetry.md:76]:
```suggestion
We also provide a [complete sample application](examples/AWSDriverExample/src/main/java/software/amazon/TelemetryMetricsOTLPExample.java) using telemetry in the examples folder of this repository.
```

**Review by karenc-bq — APPROVED** [review]

**Pull request merged**